### PR TITLE
Pj 12 login

### PR DESCRIPTION
**Pull request by NykoEugen**

### CODE REVIEW

**Review by acman — CHANGES_REQUESTED** [review]
Lets wait with this PR until registration and crispy forms will be merged. Too much changes are not related to login functionality.

After that prs are merged in main, rebase it from main and only login changes will be here.

**Review by acman — APPROVED** [review]
Looks nice! But seems tests fail

File: users/tests.py [hunk 1]
```diff
@@ -50,3 +50,53 @@ def test_signup_view_post_invalid_data(self):
 
         # Check that user are not on db
         self.assertEqual(get_user_model().objects.count(), 0)
+
+
+class LogInViewTest(TestCase):
+    def setUp(self):
+        # Test client
+        self.client = Client()
```
Comment (acman) [inline users/tests.py]: I think, u can use self.client by default from TestCase, it is not required to set it to Client()

Comment (NykoEugen) [reply users/tests.py]: I Check it

**Pull request merged**